### DISCUSSION TITLE
Command argument that uses the current input to search

### DISCUSSION THREAD

**Discussion by cody-quinn** [Minestom/Minestom#932] [Help] [answered]
I want to make a command that will use the currently provided argument to search a database and use that for auto complete results, how can I do this?

**Comment by GoldenStack** [comment, marked as answer]
You can use `Argument#setSuggestionCallback(SuggestionCallback)` and add `SuggestionEntry` instances to the `Suggestion` that is provided.
You should be able to get the context of arguments in the command that were already parsed by using `CommandContext#get` on the context that is provided in the SuggestionCallback.

For example:
```java
exampleArgument.setSuggestionCallback((sender, context, suggestion) -> {
    var result = context.get("previous_argument"); // Get data from another argument
    suggestion.addEntry(new SuggestionEntry("some suggestion", Component.text("some tooltip"))); // Add a suggestion (tooltip component is optional)
});
```
